extensions/ql-vscode/src/config.ts

```diff
@@ -11,9 +11,10 @@ export class Setting {
   name: string;
   parent?: Setting;
 
-  constructor(name: string, parent?: Setting) {
+  constructor(name: string, parent?: Setting, public readonly isHidden = false) {
```

Comment:
What's the definition of a hidden config setting?

I think the set that are visible to users are the ones listed under `contributes.configuration.properties` in the `package.json`. I believe this list is:
```
codeQL.cli.executablePath
codeQL.logInsights.joinOrderWarningThreshold
codeQL.queryHistory.format
codeQL.queryHistory.ttl
codeQL.resultsDisplay.pageSize
codeQL.runningQueries.autoSave
codeQL.runningQueries.cacheSize
codeQL.runningQueries.customLogDirectory
codeQL.runningQueries.debug
codeQL.runningQueries.maxPaths
codeQL.runningQueries.maxQueries
codeQL.runningQueries.memory
codeQL.runningQueries.numberOfThreads
codeQL.runningQueries.quickEvalCodelens
codeQL.runningQueries.saveCache
codeQL.runningQueries.timeout
codeQL.runningTests.additionalTestArguments
codeQL.runningTests.numberOfThreads
codeQL.telemetry.enableTelemetry
codeQL.telemetry.logTelemetry
codeQL.variantAnalysis.controllerRepo
codeQL.variantAnalysis.repositoryLists
```

And the list of all settings defined in this file is:
```
codeQL.cli.executablePath
codeQL.logInsights.joinOrderWarningThreshold
codeQL.queryHistory.format
codeQL.queryHistory.ttl
codeQL.resultsDisplay.pageSize
codeQL.runningQueries.autoSave
codeQL.runningQueries.cacheSize
codeQL.runningQueries.customLogDirectory
codeQL.runningQueries.debug
codeQL.runningQueries.maxPaths
codeQL.runningQueries.maxQueries
codeQL.runningQueries.memory
codeQL.runningQueries.numberOfThreads
codeQL.runningQueries.quickEvalCodelens
codeQL.runningQueries.saveCache
codeQL.runningQueries.timeout
codeQL.runningTests.additionalTestArguments
codeQL.runningTests.numberOfThreads
codeQL.telemetry.enableTelemetry
codeQL.telemetry.logTelemetry
codeQL.variantAnalysis.controllerRepo
codeQL.variantAnalysis.repositoryLists
```

So by this definition the set of "hidden" settings is:
```
codeQL.astViewer.disableCache
codeQL.canary
codeQL.canaryQueryServer
codeQL.cli.includePrerelease
codeQL.cli.personalAccessToken
codeQL.variantAnalysis.actionBranch
codeQL.variantAnalysis.liveResults
codeQL.variantAnalysis.repositoryListsPath
telemetry.enableTelemetry
```

Reply:
Thank you for a detailed look here. You're right. I should be making the settings above "hidden". Now that I think about it, I can do this more precisely and compare the settings defined in the config module with those defined in package.json. This way, we won't need to remember to mark config settings as hidden. It will happen programmatically.